pkg/hubble/filters/labels_test.go

```diff
@@ -441,6 +578,13 @@ func Test_parseSelector(t *testing.T) {
 			},
 			want: "any.dash-label.com,k8s.io.cilium in (is-awesome,rocks)",
 		},
+		{
+			name: "more complex labels",
+			args: args{
+				selector: "any:dash-label.com,k8s:io.cilium in (is-awesome, andsoon, rocks), !foobar, io.cilium notin (), test:label.com/foobar",
+			},
+			want: "any.dash-label.com,!any.foobar,any.io.cilium notin (),k8s.io.cilium in (andsoon,is-awesome,rocks),test.label.com/foobar",
```

Comment:
nit: I'm guessing the re-ordering and whitespace trimming is related to `k8sLabels.Parse()`, not `translateSelector()`. If correct can we add a small comment here?
```suggestion
			// NOTE: re-ordering and whitespace trimming is due to k8sLabels.Parse()
			want: "any.dash-label.com,!any.foobar,any.io.cilium notin (),k8s.io.cilium in (andsoon,is-awesome,rocks),test.label.com/foobar",
```